scipy/stats/stats.py

```diff
@@ -2306,6 +2311,186 @@ def zmap(scores, compare, axis=0, ddof=0):
+        This optional parameter specifies the interpolation method to use
+        when the percentile boundaries lie between two data points `i` and `j`:
+            * linear: `i + (j - i) * fraction`, where `fraction` is the
+              fractional part of the index surrounded by `i` and `j`.
+            * lower: `i`.
+            * higher: `j`.
+            * nearest: `i` or `j` whichever is nearest.
+            * midpoint: (`i` + `j`) / 2.
+    keepdims : bool, optional
+        If this is set to `True`, the reduced axes are left in the result as
+        dimensions with size one. With this option, the result will broadcast
+        correctly against the original array `x`.
+
+    Returns
+    -------
+    iqr : scalar or ndarray
+        If ``axis=None``, a scalar is returned. If the input contains integers
+        or floats of smaller precision than 64, then the output data-type is
+        ``np.float64``. Otherwise, the output data-type is the same as that of
+        the input.
+
+    References
+    ----------
+    .. [1] "Interquatrile range" https://en.wikipedia.org/wiki/Interquartile_range
+    .. [2] "Interquatrile range" http://www.stattrek.com/statistics/dictionary.aspx?definition=Interquartile%20range
+    .. [3] "Robust measures of scale" https://en.wikipedia.org/wiki/Robust_measures_of_scale
+    .. [4] "Quantile" https://en.wikipedia.org/wiki/Quantile
+
+    See Also
+    --------
+    std, var
+
+    Examples
+    --------
+    >>> from scipy.stats import iqr
+    >>> x = np.array([[10, 7, 4], [3, 2, 1]])
+    >>> x
+    array([[10,  7,  4],
+           [ 3,  2,  1]])
+    >>> iqr(x)
+    4.0
+    >>> iqr(x, axis=0)
+    array([ 3.5,  2.5,  1.5])
+    >>> iqr(x, axis=1)
+    array([ 3.,  1.])
+    >>> iqr(x, axis=1, keepdims=True)
+    array([[ 3.],
+           [ 1.]])
+
+    """
+    x = asarray(x)
+
+    # Possibly throw an exception here, so fail-fast, before lengthy computation
+    if isinstance(scale, string_types):
+        scale = _iqr_scale_to_num(scale)
+
+    # Select the percentile function to use based on nans and policy
+    contains_nan, nan_policy = _contains_nan(x, nan_policy)
+    if contains_nan and nan_policy == 'omit':
+        percentile_func = _iqr_nanpercentile
+    else:
+        percentile_func = np.percentile
+
+    # This check prevents percentile from raising an error later. Also, it is
+    # the right thing to do.
+    if not x.size: return 0.0
+
+    if len(range) != 2:
+        raise TypeError("quantile range must be two element sequence")
+
+    # reverse is True because the original computation was a one-liner:
+    #    return np.subtract(*percentile_func(x, sorted(range, reverse=True), ...), out=out)
+    # The current version is a little less obfscated
+    range = sorted(range, reverse=True)
+
+    pct = percentile_func(x, range, axis=axis,
+                          interpolation=interpolation, keepdims=keepdims)
+    out = np.subtract(pct[0], pct[1])
+
+    if scale != 1.0:
```

Comment:
Wouldn't that cause an unnecessary slowdown for the default case for large data sets?